### PR TITLE
Add waiting for two dhcp ports of a subnet in Neutron SDN

### PR DESCRIPTION
**Pull request by morozovalekseywot**

### CODE REVIEW

File: vkcs/networking/subnet.go [hunk 1]
```diff
@@ -21,15 +22,47 @@ type subnetExtended struct {
...
+		}
+
+		var allPorts []ports.Port
+		err = ports.ExtractPortsInto(allPages, &allPorts)
+		if err != nil {
+			return nil, "", fmt.Errorf("error reading VKCS Networking API response: %s", err)
+		}
+
+		if len(allPorts) < 2 {
+			return subnet, "WAITING_DHCP_PORTS", nil
```
Comment (paaanic) [inline vkcs/networking/subnet.go:64]: You should add this status to list of expected statuses or you will get an error, don't you? And do we have such test for creating a neutron subnet with enabled dhcp?

File: vkcs/networking/subnet.go [hunk 2]
```diff
@@ -21,15 +22,47 @@ type subnetExtended struct {
...
+				},
+			},
+		}
+
+		allPages, err := ports.List(client, listOpts).AllPages()
+		if err != nil {
+			return nil, "", fmt.Errorf("error listing ports of subnet: %s", err)
+		}
+
+		var allPorts []ports.Port
```
[the remark this answers is not in the document]
Comment (morozovalekseywot) [reply vkcs/networking/subnet.go:57]: may be make a struct with single field: id? pagination.Page do not have simple method to extract len

File: vkcs/networking/subnet.go (hunk 1 above)
Comment (morozovalekseywot) [reply vkcs/networking/subnet.go:64]: we do not specify pending statuses for subnet, because of this it's working. Tests are passed + i test this locally

Comment (morozovalekseywot) [reply vkcs/networking/subnet.go:64]: dhcp is enable by default, all our tests will test it

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ description: |-
 # VKCS Provider's changelog
 
 #### v0.8.5 (unreleased)
+- Add waiting for two dhcp ports of a vkcs_networking_subnet in Neutron SDN
```
[the remark this answers is not in the document]
Comment (morozovalekseywot) [reply CHANGELOG.md]: I tried to reformulate it, look at it

File: vkcs/networking/subnet.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (paaanic) [reply vkcs/networking/subnet.go]: No need, just asked if there is another way, maybe some method.

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,7 @@ description: |-
 # VKCS Provider's changelog
 
 #### v0.8.5 (unreleased)
+- Waiting for the active status of the subnet's dhcp ports
```
Comment (paaanic) [inline CHANGELOG.md:11]: I just don't like being specific about DHCP ports at all. Let's try to specify what problem you fix, not the way it's done. I suppose the problem is formulated as not actually configured DHCP service when subnet is returned as active?  Thus, we wait for it to become ready when creating subnet.

Comment (paaanic) [reply CHANGELOG.md:11]: And it's better to start all bugfix changes with "fix".

**Review by paaanic — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (morozovalekseywot) [reply CHANGELOG.md]: done

**Review by paaanic — APPROVED** [review]

**Pull request merged**